commit 6d13ae368b1ac4897da7ab9832aabe1f38147a4e
Author: Will Schmidt <will_schmidt@vnet.ibm.com>
Date:   Thu Jun 22 17:28:20 2017 -0500

    PAPI power9 event list presets
    
    Here is an initial set of events and changes to help support Power9.
    
    This is based on similar changes that were made for power8 when
    initial support was added there.  I've updated the event names to
    match what we expect to have in power9, and have done compile/build/
    sniff tests.
    
    Signed-off-by: Will Schmidt <will_schmidt@vnet.ibm.com>

diff --git a/src/linux-memory.c b/src/linux-memory.c
index 1686299..a74c866 100644
--- a/src/linux-memory.c
+++ b/src/linux-memory.c
@@ -619,7 +619,8 @@ ppc64_get_memory_info( PAPI_hw_info_t * hw_info )
 	case 0x3F:				 /* POWER7 */
 		index = 3;
 		break;
-	case 0x4b:				 /*POWER8*/
+	case 0x4b:				 /* POWER8 */
+	case 0x4e:				 /* POWER9 */
 		index = 4;
 		break;
 	default:
diff --git a/src/papi_events.csv b/src/papi_events.csv
index 4e24cfc..4385386 100644
--- a/src/papi_events.csv
+++ b/src/papi_events.csv
@@ -1516,6 +1516,50 @@ PRESET,PAPI_SR_INS,NOT_DERIVED,PM_ST_FIN
 #n/aPRESET,PAPI_BR_PRC,NOT_DERIVED,PM_BR_PRED
 #n/aPRESET,PAPI_FXU_IDL,NOT_DERIVED,PM_FXU_IDLE
 #
+CPU,POWER9
+CPU,power9
+#
+PRESET,PAPI_L1_DCM,DERIVED_ADD,PM_LD_MISS_L1,PM_ST_MISS_L1
+PRESET,PAPI_L1_LDM,NOT_DERIVED,PM_LD_MISS_L1
+PRESET,PAPI_L1_STM,NOT_DERIVED,PM_ST_MISS_L1
+PRESET,PAPI_L1_DCW,DERIVED_SUB,PM_ST_FIN,PM_ST_MISS_L1
+PRESET,PAPI_L1_DCR,DERIVED_SUB,PM_LD_REF_L1,PM_LD_MISS_L1
+#PRESET,PAPI_L1_DCA,DERIVED_POSTFIX,N0|N1|-|N2|+|N3|-,PM_ST_FIN,PM_ST_MISS_L1,PM_LD_REF_L1,PM_LD_MISS_L1
+PRESET,PAPI_L1_DCA,DERIVED_ADD,PM_LD_REF_L1,PM_ST_CMPL
+PRESET,PAPI_L2_DCM,NOT_DERIVED,PM_DATA_FROM_L2MISS
+PRESET,PAPI_L2_LDM,NOT_DERIVED,PM_L2_LD_MISS
+PRESET,PAPI_L2_STM,NOT_DERIVED,PM_L2_ST_MISS
+PRESET,PAPI_L3_DCR,NOT_DERIVED,PM_DATA_FROM_L2MISS
+PRESET,PAPI_L3_DCM,DERIVED_ADD,PM_DATA_FROM_LMEM,PM_DATA_FROM_RMEM
+PRESET,PAPI_L3_LDM,DERIVED_ADD,PM_DATA_FROM_LMEM,PM_DATA_FROM_RMEM
+PRESET,PAPI_L1_ICH,NOT_DERIVED,PM_INST_FROM_L1
+PRESET,PAPI_L1_ICM,NOT_DERIVED,PM_L1_ICACHE_MISS
+PRESET,PAPI_L2_ICM,NOT_DERIVED,PM_INST_FROM_L2MISS
+PRESET,PAPI_L2_ICM,NOT_DERIVED,PM_L2_INST_MISS
+PRESET,PAPI_L2_ICH,NOT_DERIVED,PM_INST_FROM_L2
+PRESET,PAPI_L3_ICA,NOT_DERIVED,PM_INST_FROM_L2MISS
+PRESET,PAPI_L3_ICH,NOT_DERIVED,PM_INST_FROM_L3
+PRESET,PAPI_L3_ICM,NOT_DERIVED,PM_INST_FROM_L3MISS
+PRESET,PAPI_FMA_INS,NOT_DERIVED,PM_FMA_CMPL
+PRESET,PAPI_TOT_IIS,NOT_DERIVED,PM_INST_DISP
+PRESET,PAPI_TOT_INS,NOT_DERIVED,PM_INST_CMPL
+PRESET,PAPI_INT_INS,NOT_DERIVED,PM_FXU_FIN
+PRESET,PAPI_FP_OPS,NOT_DERIVED,PM_FLOP_CMPL
+PRESET,PAPI_FP_INS,NOT_DERIVED,PM_FLOP_CMPL
+PRESET,PAPI_DP_OPS,NOT_DERIVED,PM_DP_QP_FLOP_CMPL
+PRESET,PAPI_SP_OPS,NOT_DERIVED,PM_SP_FLOP_CMPL
+PRESET,PAPI_TOT_CYC,NOT_DERIVED,PM_RUN_CYC
+PRESET,PAPI_HW_INT,NOT_DERIVED,PM_EXT_INT
+PRESET,PAPI_STL_ICY,DERIVED_POSTFIX,N0|N1|-|,PM_RUN_CYC,PM_1PLUS_PPC_DISP
+PRESET,PAPI_SR_INS,NOT_DERIVED,PM_ST_FIN
+PRESET,PAPI_LD_INS,DERIVED_ADD,PM_LD_REF_L1,PM_LD_MISS_L1
+PRESET,PAPI_LST_INS,NOT_DERIVED,PM_LSU_FIN
+PRESET,PAPI_LST_INS,DERIVED_ADD,PM_LD_REF_L1,PM_LD_MISS_L1,PM_ST_FIN
+PRESET,PAPI_BR_INS,NOT_DERIVED,PM_BRU_FIN
+PRESET,PAPI_BR_MSP,NOT_DERIVED,PM_TAKEN_BR_MPRED_CMPL
+PRESET,PAPI_BR_PRC,NOT_DERIVED,PM_BR_PRED
+PRESET,PAPI_FXU_IDL,NOT_DERIVED,PM_FXU_IDLE
+#
 CPU,ultra12
 #
 PRESET,PAPI_TOT_CYC,NOT_DERIVED,CYCLE_CNT
